### PR TITLE
 [eyehandcal] Wrist camera support 

### PR DESCRIPTION
**Pull request by 0wu**
# Description

Support a new setting for eyehand calibration: wrist camera calibration using a fixed marker in robot base frame.

* Add a new command line option `--proj-func {hand_marker_proj_world_camera,world_marker_proj_hand_camera}` to specify the `world_marker_proj_hand_camera` use case.
* The output calibration.json now contain 
  * a "proj_func" field indicating the setting used during calibration
  * a "pixel_error" field indicating the quality of the calibration for this camera

* Add testcases for collect_data_and_cal.py from realworld data.

## Sample execution result
```
(eyehandcal) tingfan@perception:~/github/fairo/perception/sandbox/eyehandcal$ collect_data_and_cal.py  --datafile caldataNew.pkl --marker-id 1 --proj-func world_marker_proj_hand_camera
Config: Namespace(calibration_file='calibration.json', datafile='caldataNew.pkl', imagedir=None, ip='100.96.135.68', marker_id=1, num_points=20, overwrite=False, pixel_tolerance=2.0, points_file='calibration_points.json', proj_func='world_marker_proj_hand_camera', seed=0, time_to_go=3)
Warning: datafile caldataNew.pkl already exists. Loading data instead of collecting data...
Done. Data has 100 poses.
Solve camera 0/1 pose
number of images with keypoint 99
stage 1 mean pixel error 28.687245439629173
stage 2 mean pixel error 1.2076883491182768
Good solution 1.2076883491182768 <= 2.0
Camera 0 calibration: {'proj_func': 'world_marker_proj_hand_camera', 'camera_ee_ori_rotvec': [-0.0533591361279038, -0.02723116674277641, 0.8119656021495493], 'camera_ee_pos': [0.03316384837244427, -0.08234773892713702, 0.027100460893550952], 'marker_base_pos': [0.5556270121252427, 0.0012146942266407848, 0.003037030620487715]}
Saving calibrated parameters to calibration.json
```

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [ ] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

If this PR introduces a change or fixes a bug, please add before and after screenshots (if this is causes a UX change) or before and after examples(this could be plain text, videos etc ) to demonstrate the change.

If this PR introduces a new feature, please also add examples or screenshots demonstrating the feature.

# Testing

Please describe the tests that you ran to verify your changes and how we can reproduce.

# Checklist:

- [ ] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

File: perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py [hunk 1]
```diff
@@ -208,6 +220,8 @@ def extract_obs_data_std(data, camera_index):
             print('stage 2 mean pixel error', pixel_error)
             if pixel_error > args.pixel_tolerance:
```
Comment (Jdvakil) [inline perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py:221]: if float(pixel_error) > float(args.pixel_tolerance):

File: perception/sandbox/eyehandcal/readme.md [hunk 2]
```diff
@@ -77,7 +83,8 @@ Run `collect_data_and_cal.py` to run the calibration process by sampling from th
 
 ```bash
 $ collect_data_and_cal.py --help
-usage: collect_data_and_cal.py [-h] [--ip IP] [--datafile DATAFILE] [--overwrite] [--marker-id MARKER_ID] [--calibration-file CALIBRATION_FILE] [--points-file POINTS_FILE] [--num-points NUM_POINTS] [--time-to-go TIME_TO_GO] [--imagedir IMAGEDIR]
+usage: collect_data_and_cal.py [-h] [--seed SEED] [--ip IP] [--datafile DATAFILE] [--overwrite] [--marker-id MARKER_ID] [--calibration-file CALIBRATION_FILE] [--points-file POINTS_FILE] [--num-points NUM_POINTS]
```
Comment (Jdvakil) [inline perception/sandbox/eyehandcal/readme.md:86]: --proj-func arguments could have more intuitive names like wristCam and steadCam.

File: perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py (hunk 1 above)
Comment (0wu) [reply perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py:221]: arg.pixel_error is now float as we added `type=float` in https://github.com/facebookresearch/fairo/pull/1113/files#diff-9c0bbc45d239fbb8bb96cb7e742d0ee57f2395e4eb9948f1d833cb5039e24072R130

Comment (0wu) [reply perception/sandbox/eyehandcal/scripts/collect_data_and_cal.py:221]: arg.pixel_error is now float as we added `type=float` in https://github.com/facebookresearch/fairo/pull/1113/files#diff-9c0bbc45d239fbb8bb96cb7e742d0ee57f2395e4eb9948f1d833cb5039e24072R130

**Review by Jdvakil — APPROVED** [review]

**Pull request merged**